utility/mm2_crawler

```diff
@@ -126,12 +151,45 @@ def doit(options, config):
     # not totally sure why...
     os.chdir('/var/tmp')
 
+    signal.signal(signal.SIGALRM, sigalrm_handler)
     # Then create a threadpool to handle as many at a time as we like
     threadpool = multiprocessing.pool.ThreadPool(processes=options.threads)
     fn = lambda host_id: worker(options, config, host_id)
 
     # Here's the big operation
-    return_codes = threadpool.map(fn, hosts)
+    result = threadpool.map_async(fn, hosts)
+
+    while not result.ready():
+        time.sleep(1)
+        if shutdown:
+            # Being here means that the signal handler was called and the
+            # timeout was decreased to -1. Not starting any new workers.
+            threadpool.close()
+            # Wait for SHUTDOWN_TIMEOUT minutes for all threads to shutdown
+            shutdown_timer = time.time()
+            shutdown = False
+        if shutdown_timer:
+            delta = time.time() - shutdown_timer
+            if delta > SHUTDOWN_TIMEOUT * 60:
+                # Time out for graceful shutdown is over.
+                # Let's terminate all threads.
+                logger.warning("About to terminate all threads.")
+                threadpool.terminate()
+                logger.warning("About to terminate all threads. Done.")
+                break
```

Comment:
ok, that's fine then.

Basically, I was wondering if ``result.ready()`` could return ``True`` while we wanted to shutdown and what was going to happen then, but if that's the case, it will just exit fine.